packages/ui/src/helpers/storage/fileUploader/utils/uploader.ts

```diff
@@ -24,3 +24,37 @@ export function uploadFile({
   // stub
   return;
 }
+
+/**
+ * Format bytes as human-readable text.
+ *
+ * @param bytes Number of bytes.
+ * @param si True to use metric (SI) units, aka powers of 1000. False to use
+ *           binary (IEC), aka powers of 1024.
+ * @param dp Number of decimal places to display.
+ *
+ * @return Formatted string.
+ */
+export function humanFileSize(bytes, si = false, dp = 1) {
+  const thresh = si ? 1000 : 1024;
+
+  if (Math.abs(bytes) < thresh) {
+    return bytes + ' B';
+  }
+
+  const units = si
+    ? ['kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+    : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+  let u = -1;
+  const r = 10 ** dp;
```

Comment:
nit, what are `u`, `r`, and `dp`? we should avoid abbreviated names for readability if possible.

Reply:
I'll update `u` and `r`, `dp` though is declared as decimal places in the comments above.